src/analysis/mod.rs

```diff
@@ -83,12 +92,65 @@ pub fn run(env: &mut Env) {
         return;
     }
 
+    analyze_enums(env);
+
+    analyze_flags(env);
+
     analyze_constants(env);
 
     // Analyze free functions as the last step once all types are analyzed
     analyze_global_functions(env);
 }
 
+fn analyze_enums(env: &mut Env) {
+    let mut imports = Imports::new(&env.library);
+    imports.add("glib::translate::*");
+    imports.add(&format!("crate::{}", env.main_sys_crate_name()));
+
+    for obj in env.config.objects.values() {
+        if obj.status.ignored() {
+            continue;
+        }
+        let tid = match env.library.find_type(0, &obj.name) {
+            Some(x) => x,
+            None => continue,
+        };
+
+        if let Type::Enumeration(_) = env.library.type_(tid) {
+            if let Some(info) = enums::new(env, obj, &mut imports) {
+                env.analysis.enumerations.push(info);
+            }
+        }
+    }
+
+    env.analysis.enum_imports = imports;
+}
+
+fn analyze_flags(env: &mut Env) {
+    let mut imports = Imports::new(&env.library);
+    imports.add("glib::translate::*");
+    imports.add(&format!("crate::{}", env.main_sys_crate_name()));
```

Comment:
This is actually not added to the generated code but filtered out again later, isn't it?

Reply:
You're right, all code was changed for 2018 to use `ffi::something` directly but all these calls were still left in the other analysis/codegen files. I've removed all these `.add` calls, as well as turn the `name == "crate::ffi"` check in `common_checks` into an assertion such that any future mistake like this is more easily caught.

Reply:
Note that this also removes the following from gtk-rs:

```diff
diff --git a/glib/src/gobject/auto/binding.rs b/glib/src/gobject/auto/binding.rs
index 1765250fd..d219724b3 100644
--- a/glib/src/gobject/auto/binding.rs
+++ b/glib/src/gobject/auto/binding.rs
@@ -2,7 +2,6 @@
 // from gir-files (https://github.com/gtk-rs/gir-files)
 // DO NOT EDIT
 
-use crate::gobject_ffi;
 use crate::translate::*;
 use crate::BindingFlags;
 use std::fmt;
diff --git a/glib/src/gobject/auto/flags.rs b/glib/src/gobject/auto/flags.rs
index e1fbed625..670c941e8 100644
--- a/glib/src/gobject/auto/flags.rs
+++ b/glib/src/gobject/auto/flags.rs
@@ -2,7 +2,6 @@
 // from gir-files (https://github.com/gtk-rs/gir-files)
 // DO NOT EDIT
 
-use crate::gobject_ffi;
 use crate::translate::*;
 use crate::value::FromValue;
 use crate::value::FromValueOptional;
```

Reply:
Nice :)